src/server/scripts/Kalimdor/RagefireChasm/boss_adarogg.cpp

```diff
@@ -0,0 +1,223 @@
+/*
+ * This file is part of the TrinityCore Project. See AUTHORS file for Copyright information
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "InstanceScript.h"
+#include "MotionMaster.h"
+#include "Player.h"
+#include "ScriptMgr.h"
+#include "ScriptedCreature.h"
+#include "TaskScheduler.h"
+#include "ragefire_chasm.h"
+
+enum Spells
+{
+    // Intro
+    SPELL_ADJULES_CHOW_TIME         = 120113,
+    SPELL_EAT_TROGG                 = 120064,
+
+    SPELL_INFERNO_CHARGE_CAST       = 119405,
+    SPELL_INFERNO_CHARGE_SUMMON     = 119297, // Serverside
+    SPELL_FIRE_BREATH               = 119420
+};
+
+enum Texts
+{
+    SAY_INFERNO_CHARGE              = 0,
+
+    // Intro
+    SAY_HOUNDMASTER_INTRO_0         = 0,
+    SAY_HOUNDMASTER_INTRO_1         = 1
+};
+
+enum Events
+{
+    EVENT_INFERNO_CHARGE            = 1,
+    EVENT_FIRE_BREATH               = 2
+};
+
+enum Actions
+{
+    ACTION_KILL_HOUNDMASTERS        = 1
+};
+
+enum Waypoints
+{
+    // Intro Waypoint
+    PATH_INTRO                      = 6140800,
+
+    POINT_EAT_1                     = 0,
+    POINT_EAT_2                     = 3
+};
+
+constexpr Position IntroPoints[2] =
+{
+    { -282.31488f, -53.24906f, -60.802902f },
+    { -281.1894f,  -54.73433f, -60.34256f }
+};
+
+// Areatrigger - 7904
+class at_adarogg_intro : public OnlyOnceAreaTriggerScript
+{
+public:
+    at_adarogg_intro() : OnlyOnceAreaTriggerScript("at_adarogg_intro") { }
+
+    bool TryHandleOnce(Player* player, AreaTriggerEntry const* /*areaTrigger*/) override
+    {
+        if (InstanceScript* instance = player->GetInstanceScript())
+            if (Creature* adarogg = instance->GetCreature(BOSS_ADAROGG))
+                adarogg->AI()->DoAction(ACTION_KILL_HOUNDMASTERS);
+
+        return true;
+    }
+};
+
+// 61408 - Adarogg
+struct boss_adarogg : public BossAI
+{
+    using BossAI::BossAI;
```

Comment:
drop this line, you gotta explicitly call the ctor of BossAI with `DATA_ADAROGG` as param, if you wouldn't need any specific params you could use the using (like for ScriptedAI, but not here)